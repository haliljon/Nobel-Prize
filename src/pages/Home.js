import React from 'react';
import { NavLink } from 'react-router-dom';
import chemistry from '../assets/chemistry.png';
import literature from '../assets/literature.png';
import physics from '../assets/physics.png';
import economics from '../assets/economics.png';
import medicine from '../assets/medicine.png';
import peace from '../assets/peace.png';

const Home = () => {
  const categories = [
    { name: 'chemistry', icon: chemistry },
    { name: 'economics', icon: economics },
    { name: 'literature', icon: literature },
    { name: 'physics', icon: physics },
    { name: 'medicine', icon: medicine },
    { name: 'peace', icon: peace }];
  const [searchTerm, setSearchTerm] = React.useState([]);
  const handleChange = (event) => {
    setSearchTerm(event.target.value);
  };
  const filteredCategories = categories.filter(
    (category) => category.name.toLowerCase().includes(searchTerm),
  );
  return (
    <div>
      <div className="search-bar">
        <input className="search" type="text" placeholder="Search for Category" onChange={handleChange} />
      </div>
      <div className="nobelImg container mt-3 mb-3">
        <img className="nobel-logo" src="https://upload.wikimedia.org/wikipedia/en/e/ed/Nobel_Prize.png" alt="nobel-logo" />
        <div>The Nobel Prize Winners</div>
      </div>
      <div className="heading pt-3">
        <p className="ps-4">NOBEL PRIZE NOMINEES by CATEGORY</p>
      </div>
      <div className="container mt-3 mb-3">
        <div className="row">
          {filteredCategories.map((category) => (
            <NavLink className="col-4 btn eachCateg" key={category.name} to={`/category/${category.name}`}>
              <div className="category">
                <div className="category-icon">
                  <img className="img-fluid" src={category.icon} alt={category.name} />
                </div>
                <div>
                  <h3>{category.name.charAt(0).toUpperCase() + category.name.slice(1)}</h3>
                </div>
              </div>
            </NavLink>
          ))}
        </div>
      </div>
    </div>
  );
};

export default Home;
