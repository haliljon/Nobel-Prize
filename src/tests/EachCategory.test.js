import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../pages/Home';
import store from '../redux/configureStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
