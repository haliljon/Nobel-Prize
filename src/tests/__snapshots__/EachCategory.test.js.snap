// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="search-bar"
  >
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search for Category"
      type="text"
    />
  </div>
  <div
    className="nobelImg container mt-3 mb-3"
  >
    <img
      alt="nobel-logo"
      className="nobel-logo"
      src="https://upload.wikimedia.org/wikipedia/en/e/ed/Nobel_Prize.png"
    />
    <div>
      The Nobel Prize Winners
    </div>
  </div>
  <div
    className="heading pt-3"
  >
    <p
      className="ps-4"
    >
      NOBEL PRIZE NOMINEES by CATEGORY
    </p>
  </div>
  <div
    className="container mt-3 mb-3"
  >
    <div
      className="row"
    >
      <a
        className="col-6 col-sm-6 col-lg-4 btn eachCateg"
        href="/category/chemistry"
        onClick={[Function]}
      >
        <div
          className="category"
        >
          <div
            className="category-icon"
          >
            <img
              alt="chemistry"
              className="img-fluid"
              src="chemistry.png"
            />
          </div>
          <div>
            <h3>
              Chemistry
            </h3>
          </div>
        </div>
      </a>
      <a
        className="col-6 col-sm-6 col-lg-4 btn eachCateg"
        href="/category/economics"
        onClick={[Function]}
      >
        <div
          className="category"
        >
          <div
            className="category-icon"
          >
            <img
              alt="economics"
              className="img-fluid"
              src="economics.png"
            />
          </div>
          <div>
            <h3>
              Economics
            </h3>
          </div>
        </div>
      </a>
      <a
        className="col-6 col-sm-6 col-lg-4 btn eachCateg"
        href="/category/literature"
        onClick={[Function]}
      >
        <div
          className="category"
        >
          <div
            className="category-icon"
          >
            <img
              alt="literature"
              className="img-fluid"
              src="literature.png"
            />
          </div>
          <div>
            <h3>
              Literature
            </h3>
          </div>
        </div>
      </a>
      <a
        className="col-6 col-sm-6 col-lg-4 btn eachCateg"
        href="/category/physics"
        onClick={[Function]}
      >
        <div
          className="category"
        >
          <div
            className="category-icon"
          >
            <img
              alt="physics"
              className="img-fluid"
              src="physics.png"
            />
          </div>
          <div>
            <h3>
              Physics
            </h3>
          </div>
        </div>
      </a>
      <a
        className="col-6 col-sm-6 col-lg-4 btn eachCateg"
        href="/category/medicine"
        onClick={[Function]}
      >
        <div
          className="category"
        >
          <div
            className="category-icon"
          >
            <img
              alt="medicine"
              className="img-fluid"
              src="medicine.png"
            />
          </div>
          <div>
            <h3>
              Medicine
            </h3>
          </div>
        </div>
      </a>
      <a
        className="col-6 col-sm-6 col-lg-4 btn eachCateg"
        href="/category/peace"
        onClick={[Function]}
      >
        <div
          className="category"
        >
          <div
            className="category-icon"
          >
            <img
              alt="peace"
              className="img-fluid"
              src="peace.png"
            />
          </div>
          <div>
            <h3>
              Peace
            </h3>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
